aws_topology/stackstate_checks/aws_topology/cloudtrail.py

```diff
@@ -0,0 +1,126 @@
+import json
+import dateutil.parser
+import pytz
+from datetime import datetime
+import gzip
+import io
+import botocore
+from six import string_types, PY2
+
+
+try:
+    JSONParseException = json.decoder.JSONDecodeError
+except AttributeError:  # Python 2
+    JSONParseException = ValueError
+
+
+class CloudtrailCollector(object):
+    def __init__(self, bucket_name, account_id, session, agent):
+        self.bucket_name = bucket_name
+        self.session = session
+        self.agent = agent
+        self.account_id = account_id
+
+    def get_messages(self, not_before):
+        try:
+            # try s3
+            return self._get_messages_from_s3(not_before)
+        except Exception:
+            # try loopup_events
+            self.agent.warning("Falling back to slower Cloudtrail lookup_events")
+            return self._get_messages_from_cloudtrail(not_before)
```

Comment:
Maybe you should wrap this in a try as well or what Exception do you expect from `_get_messages_from_s3`?

Reply:
Ah I guess it doesn't matter, you want to escalate it. Disregard this one